src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.AbstractGlobalSuppressMessageCodeAction.cs

```diff
@@ -70,12 +70,13 @@ protected async Task<Document> GetOrCreateSuppressionsDocumentAsync(Cancellation
                         var fullPath = !string.IsNullOrEmpty(filePath) ? Path.GetFullPath(filePath) : filePath;
                         if (fullPath == suppressionsFilePath)
                         {
-                            // Existing global suppressions file, see if this file only has global assembly attributes.
+                            // Existing global suppressions file. See if this file only has imports and global assembly
+                            // attributes.
                             hasDocWithSuppressionsName = true;
 
                             var t = await document.GetSyntaxTreeAsync(c).ConfigureAwait(false);
                             var r = await t.GetRootAsync(c).ConfigureAwait(false);
-                            if (r.ChildNodes().All(n => Fixer.IsAttributeListWithAssemblyAttributes(n)))
+                            if (r.ChildNodes().All(n => Fixer.IsImportsLine(n) || Fixer.IsAttributeListWithAssemblyAttributes(n)))
```

Comment:
you have a _project.  so you can get the ISyntaxFactsService for it, and then just use IsUsingOrExternOrImport.

Reply:
This will also just address @sharwell 's issue automatically, which is a plus :)

Reply:
Note: i also don't think we need IsAttributeListWithAssemblyAttributes we have ISyntaxFactsIsGlobalAttribute that can be used instead.  Though we'd need to add an ISyntaxFacts.IsAttributeList as well.

Reply:
@CyrusNajmabadi 

> Note: i also don't think we need IsAttributeListWithAssemblyAttributes we have ISyntaxFactsIsGlobalAttribute that can be used instead. Though we'd need to add an ISyntaxFacts.IsAttributeList as well.

That was there in the existing code, obviously. I'd prefer that to be done in a perpendicular PR to keep this one lean.

Reply:
@CyrusNajmabadi 

> Note: i also don't think we need IsAttributeListWithAssemblyAttributes we have ISyntaxFactsIsGlobalAttribute that can be used instead. Though we'd need to add an ISyntaxFacts.IsAttributeList as well.

That was there in the existing code, obviously. I'd prefer that to be done in a perpendicular PR to keep this one lean.

Reply:
> That was there in the existing code, obviously. I'd prefer that to be done in a perpendicular PR to keep this one lean.

I have no problem with that.  It was just a suggestion :)